src/dynimage.rs

```diff
@@ -733,7 +733,13 @@ fn open_impl(path: &Path) -> ImageResult<DynamicImage> {
         Ok(f) => f,
         Err(err) => return Err(image::ImageError::IoError(err)),
     };
-    let fin = BufReader::new(fin);
+
+    match load_guess(BufReader::new(&fin)) {
+        Ok(im) => return Ok(im),
+        Err(_e) => {
```

Comment:
I am not completely sure this is correct. What happens if there is a decoding error later in the image, wouldn't it try to read it again using the extension, and starting where-ever the first decoder stopped? You probably need to duplicate some functionality from `load_guess`?

Reply:
Not sure I follow what you mean here.
Like, if load_guess() guesses its a PNG and we return Ok(im) here, gut the png data stream is corrupt ?

Reply:
Do you mean like if load_guess() detects PNG,  but fails to decode it and load_guess() then returns an Err here,
and further down we will again try to decode this png but this time because of the file extension?

Reply:
Exactly!

Reply:
Alright, yea in the error path it would do 2x work.
I'll see if i can figure something better out.

Reply:
Ok resolved this too with a new commit.

Verified the only files who fallthru are the .tga files as expected since they're not detected by magic bytes.

Reply:
With #928 this should work, but unfortunately `UnsupportedError` is currently used for more than just unsupported formats.

Reply:
Lets pause this until #928 lands then.